announcements/announcements_test.go

```diff
@@ -0,0 +1,91 @@
+package announcements
+
+import (
+	"reflect"
+	"testing"
+)
+
+func Test_formatAnnouncement(t *testing.T) {
+	type args struct {
+		maxLen  int
+		subject string
+		body    string
+	}
+	tests := []struct {
+		name string
+		args args
+		want Announcement
+	}{
+		{
+			name: "no-cutoff",
+			args: args{
+				maxLen:  100,
+				subject: "test",
+				body:    "test",
+			},
+			want: Announcement{
+				Subject:  "test",
+				Contents: []string{"test"},
+			},
+		},
+		{
+			name: "cutoff",
+			args: args{
+				maxLen:  10,
+				subject: "test",
+				body:    "0.1.2.3.4.5.6.7.8.9",
+			},
+			want: Announcement{
+				Subject:  "test",
+				Contents: []string{"0.1.2.", "3.4.5.6.7.", "8.9"},
+			},
+		},
+		{
+			name: "cutoff that ends in a dot",
+			args: args{
+				maxLen:  11,
+				subject: "test",
+				body:    ".1.2.3.4.5.6.7.",
+			},
+			want: Announcement{
+				Subject:  "test",
+				Contents: []string{".1.2.3.", "4.5.6.7."},
+			},
+		},
+		{
+			name: "we cant split it",
+			args: args{
+				maxLen:  10,
+				subject: "test",
+				body:    "0123456789",
+			},
+			want: Announcement{
+				Subject:  "test",
+				Contents: []string{"0123456789"},
+			},
+		},
+		{
+			name: "with end delimiter",
+			args: args{
+				maxLen:  10,
+				subject: "test",
+				body: `Full body test.
+example announcement.
+It has an end delimiter. \-\-
+and then an email.
+`,
+			},
+			want: Announcement{
+				Subject:  "test",
+				Contents: []string{"Full body test.", "example announcement.", "It has an end delimiter."},
```

Comment:
The max len for this example was set to 10, but the individual components all exceed 10 characters, one is even 20 characters. Not sure if this how this should be implemented given the test cases, it's going to error out in a lot of situations it previously did if a message crosses 2000 chars mid sentence

Reply:
This will have an issue only if a message doesn't have any punctuation/newline in the last 2000 characters, which is frankly enough I think.

I just tried to make it clean enough